javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -564,8 +564,23 @@ object Integer {
   @inline def toUnsignedLong(x: scala.Int): scala.Long =
     intToULong(x)
 
-  @inline def valueOf(i: scala.Int): Integer =
-    new Integer(i)
+  import IntegerCache.cache
+
+  @inline def valueOf(intValue: scala.Int): Integer = {
+    if (intValue < -128 || intValue > 127) {
```

Comment:
You can do this test with only one branch as `(intValue.toByte.toInt != intValue)`. Same for the `Short` and `Long` cases.